modules/integration/src/test/scala/scala/cli/integration/RunScalacCompatTestDefinitions.scala

```diff
@@ -206,6 +206,43 @@ trait RunScalacCompatTestDefinitions {
     }
   }
 
+  test("scala-cli compile -d option package into a jar") {
+    val preCompileDir = "PreCompileDir"
+    val preCompiledInput = "Message.scala"
+    val mainInput = "Main.scala"
+    val expectedOutput = "Hello"
+    TestInputs(
+      os.rel / preCompileDir / preCompiledInput -> "case class Message(value: String)",
+      os.rel / preCompileDir / mainInput -> s"""object Main extends App { println(Message("$expectedOutput").value) }"""
```

Comment:
minor point - it's unnecessary to have these two source files because you compile them together - its probably fine to drop the `Message` class altogether and just call println with expectedOutput